crates/solver/src/driver.rs

```diff
@@ -350,7 +350,11 @@ impl Driver {
                         error_at_earlier_block,
                     );
                     // split warning into separate logs so that the messages aren't too long.
-                    tracing::warn!("settlement failure for: \n{:#?}", settlement);
+                    tracing::warn!(
+                        "{} settlement failure for: \n{:#?}",
+                        solver.name(),
+                        settlement,
+                    );
```

Comment:
A little late to the party, but I would have suggested:
```rust
                    tracing::warn!(
                        solver_name =% solver.name(), ?settlement,
                        "settlement failure",
                    );
```

I find `solver_name=...` more greppable in the logs (and would also give you related surrounding logs).

This is 100% fine like this though - I've just become a huge fan of this tracing syntax for attaching additional data to logs (instead of writing it `std::fmt`-style).